lib/setup-lib.bash

```diff
@@ -66,6 +66,9 @@ function clobbering() {
 }
 
 function grep_or_add() {
+  if [ -n "${ASDF_DIRENV_NO_TOUCH_RC_FILE-}" ]; then
```

Comment:
This doesn't feel like the right place to put this to me. This low-level helper method is a generic file manipulation tool, it isn't specific to shell rc files.

Ideas:

1. Update all the call sites to instead bail out before calling this. Or, if that's just too messy:
2. Rename this function from `grep_or_add` to something like `maybe_add_to_shellrc` just to make it clear that it should only be used for shellrc files.

Reply:
I agree '1' would look messy. So I added a function ```maybe_add_to_shellrc()```, that is specific to shell rc files.

Reply:
I hate redundant code, and after looking again at the ```maybe_add_to_shellrc()``` function decided it was too ugly. I went with number 1, and it turned out to be much cleaner.